### PR TITLE
Add FormDefinition types, move definition JSONs to backend

### PR DESCRIPTION
**Pull request by gigxz**
* Moved all types for FormDefinition shape from frontend to backend (corresponding frontend PR: https://github.com/greenriver/hmis-frontend/pull/59)
* Moved form definition JSON to this repo. To load the definitions necessary for the frontend to work, run `rails driver:hmis:seed_definitions`
* Included FormDefinitions for "records" (e.g. form for create/edit project, funder, client, etc) as well as one dummy intake assessment definition. For now, the frontend fetches those directly by `identifier`, so they don't get `Instance` records. Instance is only used for assessments. Added `query formDefinition` for fetching definitions in this way.
* Added `query pickList` for getting list of CoCs, projects, organizations. May add more later, for picklists that depend on  backend configuration, or picklists that need to be grouped.
* Move `FormDefinition` out of `HmisSchema` namespace and into `Forms` namespace. I think to keep things clean, lets use HmisSchema namespace only for HUD HMIS defined types.
* Implement `data_collected_about` logic to filter conditionals before returning

### CODE REVIEW

File: drivers/hmis/app/models/hmis/form/definition.rb [hunk 1]
```diff
@@ -37,4 +37,36 @@ def self.find_definition_for_project(project, role:, version: nil)
...
+  end
+
+  private def matches_condition(condition, enrollment, client)
+    case condition
+    when 'ALL_CLIENTS'
+      true
+    when 'HOH'
+      enrollment.RelationshipToHoH == 1
+    when 'HOH_AND_ADULTS'
+      enrollment.RelationshipToHoH == 1 || client.age > 18
```
Comment (eanders) [inline drivers/hmis/app/models/hmis/form/definition.rb]:
```suggestion
      enrollment.RelationshipToHoH == 1 || client.age >= 18
```

File: drivers/hmis/app/models/hmis/hud/client.rb [hunk 2]
```diff
@@ -137,4 +137,20 @@ def self.sort_by_option(option)
...
+    self.class.age(date: date, dob: dob)
+  end
+
+  def self.age(date:, dob:)
+    return nil unless date.present? && dob.present?
+
+    age = date.year - dob.year
+    age -= 1 if dob > date.years_ago(age)
+    age
+  end
```
Comment (eanders) [inline drivers/hmis/app/models/hmis/hud/client.rb]: Let's just depend on the logic we already have in `GrdaWarehouse::Hud::Client`
```suggestion
  def age(date = Date.current)
    GrdaWarehouse::Hud::Client.age(date: date, dob: self.DOB)
  end

  def self.age(date:, dob:)
    GrdaWarehouse::Hud::Client.age(date: date, dob: self.DOB)
  end
```

File: drivers/hmis/lib/form_data/assessments/dummy_intake_assessment.json [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+          "pick_list_reference": "yesNoMissing"
+        },
+        {
+          "type": "GROUP",
+          "link_id": "4.09-group",
+          "item": [
+            {
+              "type": "CHOICE",
+              "link_id": "4.09-A",
+              "text": "Is this a disabling condition?",
```
Comment (eanders) [inline drivers/hmis/lib/form_data/assessments/dummy_intake_assessment.json:302]: Is this the new language for indefinite and impairing?

**Review by eanders — APPROVED** [review]

File: drivers/hmis/lib/form_data/assessments/dummy_intake_assessment.json (hunk 3 above)
Comment (gigxz) [reply drivers/hmis/lib/form_data/assessments/dummy_intake_assessment.json:302]: This file in particular isn't really ready for review. I'm working on updating it today, will have another PR later!

**Pull request merged**